src/blog/2025/04/building-oee-dashboard-with-flowfuse-part-3.md

```diff
@@ -0,0 +1,275 @@
+authors: ["sumit-shinde"]
+image: /blog/2025/04/images/building-oee-dashboard-part1.png
+keywords: free oee dashboard, oee dashboard free, building oee dashboard, node-red oee dashboard, flowfuse oee dashboard
+tags:
+   - flowfuse
+---
+
+In [Part 2](/blog/2025/04/building-oee-dashboard-with-flowfuse-2/), we built the flow to calculate OEE for the production line using simulated production and downtime data and created a dashboard interface for visualization. However, we did not focus much on theme, layout, or styling.
+
+<!--more-->
+
+In Part 3, we will focus on improving the theme and design of the OEE dashboard. We will learn how to connect a real data source, adjust fields if your data structure differs, scale the dashboard for multiple production lines, and finally, explore how you can use it to take action based on insights.
+
+Let's get started!
+
+## Enhancing the Dashboard Theme and Design
+
+In the planning section of [Part 1](/blog/2025/04/building-oee-dashboard-with-flowfuse-part-1/), we introduced a mockup of the dashboard with a modern dark theme. The theme was built around a sleek, professional aesthetic, using high-contrast colors for readability and a visually appealing layout.
+
+The primary colors in the theme include:
+
+- **Black (#000000)** — used for the page background to create contrast and reduce eye strain.
+- **Charcoal Blue (#1A1C24)** — a deep, muted tone that adds depth while maintaining a clean and modern look, used for the groups.
+- **White (#FFFFFF)** — used for text elements to ensure maximum readability against the dark background.
+- **Accent Colors** — vibrant colors such as teal, orange, green, yellow, and blue are used across widget elements, including chart bars, line graphs, and indicators. These accents help differentiate data types and bring attention to key metrics.
+
+### To update the dashboard theme in Node-RED Dashboard 2.0:
+
+1. Open the Dashboard 2.0 sidebar from the Node-RED editor.
+2. Switch to the Theme tab.
+3. In the list of themes (you will likely see only the default one), click the settings (gear) icon next to it.
+4. In the theme settings:
+    - Set Charcoal Blue (#1A1C24) as the color for the header background, group background, and group outline.
+    - Set Black (#000000) as the page background.
+5. Click **Update** and **Deploy Changes**.
+
+Your dashboard should display the updated dark theme with a clean, modern appearance and improved visual contrast.
+
+However, additional adjustments are needed to fully align the visuals, specifically the chart grid lines and label text colors.
+
+### To update these:
+
+1. Double-click on a chart widget to open its configuration panel.
+2. Scroll to the bottom of the chart config UI.
+3. Uncheck the following options:
+    - Use ChartJs Default Text Colors
+    - Use ChartJs Default Grid Colors
+4. Set the text color to `#FFFFFF` (white) and the grid line color to `#606060`.
+5. Click **Done**, then **Deploy the changes**.
```

Comment:
I get all this, but I think it's more valuable to explain how I can update the blueprint, or make a dashboard like this, where I get to pick the colors.

I wouldn't know how to pick the colors, what a good color would be for text, etc etc.